MapboxSearch/offline/src/androidTest/java/com/mapbox/search/offline/OfflineSearchEngineIntegrationTest.kt

```diff
@@ -558,17 +557,41 @@ internal class OfflineSearchEngineIntegrationTest {
         fun compareWithApproximateLocations(expected: BaseRawSearchResult, serverResult: BaseRawSearchResult): Boolean {
             if (expected == serverResult) return true
 
-            val fixedResult = expected.copy(
-                id = serverResult.id,
-                userRecordPriority = serverResult.userRecordPriority,
-                center = serverResult.center,
-                distanceMeters = serverResult.distanceMeters
-            )
-            return fixedResult == serverResult &&
-                    expected.center.approximatelyEquals(serverResult.center) &&
-                    compareDistanceMeters(expected.distanceMeters, serverResult.distanceMeters)
+            if (assertSearchResultEquals(expected, serverResult))
+                return true
+
+            Log.d(LOG_TAG, "compareWithApproximateLocations: expected = $expected, serverResult = $serverResult")
+            return false
         }
 
+        fun assertSearchResultEquals(expected: BaseRawSearchResult, actual: BaseRawSearchResult): Boolean {
+            assertEquals(expected.types, actual.types)
+            assertEquals(expected.names, actual.names)
+            assertEquals(expected.languages, actual.languages)
+            assertEquals(expected.addresses, actual.addresses)
+            assertEquals(expected.descriptionAddress, actual.descriptionAddress)
+            assertEquals(expected.matchingName, actual.matchingName)
+            assertEquals(expected.fullAddress, actual.fullAddress)
+            assertEquals(expected.routablePoints?.sortedBy { it.point.hashCode() }?.joinToString(";") { it.toString() }, actual.routablePoints?.sortedBy { it.point.hashCode() }?.joinToString(";") { it.toString() })
```

Comment:
You can pass lists to `assertEquals()` directly if elements order matters.

Reply:
I don't think that it will work since it requires an overload for `equals` which is not the case for some types...